cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -1128,6 +1133,13 @@ func (csr *ClusterStateRegistry) handleInstanceCreationErrorsForNodeGroup(
 		}
 	}
 
+	ignoreErrors, err := csr.nodeGroupConfigProcessor.GetIgnoreInstanceCreationErrors(nodeGroup)
+	if err != nil {
+		klog.V(1).Infof("Failed to find IgnoreInstanceCreationErrors for nodeGroup %v with error: %v", nodeGroup.Id(), err)
+	} else if ignoreErrors {
+		return // The scaleUp will timeout after MaxNodeProvisionTime is reached (ClusterStateRegistry::updateScaleRequests).
+	}
+
```

Comment:
We only want to ignore stockout errors, so we probably want to move it into the loop so that we know `errorCode.class`